From: KART <dev@kart>
Date: Sun, 17 Aug 2026 00:00:00 +0900
Subject: [PATCH] imx8mm: RDC grant M4 access to ECSPI2/GPIO3/GPIO5

repro-mu-read-reset (pitfalls #26) の検証用。既定の ATF RDC 設定は M4
(domain1) に UART4 しか割り当てておらず、ECSPI2/GPIO は権限テーブルに
無い。M4 が rpmsg セッション稼働中にこれらを read すると SoC がハード
リセットする現象があり、ブート時 RDC でこれらを両ドメイン RW に明示
割り当てすると直るかを試す。

D0R|D0W|D1R|D1W = 両ドメイン読み書き可 (Linux も M4 も触れる)。
---
 plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c b/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c
index 1111111..2222222 100644
--- a/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c
+++ b/plat/imx/imx8m/imx8mm/imx8mm_bl31_setup.c
@@ -120,6 +120,9 @@ static const struct imx_rdc_cfg rdc[] = {
 	RDC_PDAPn(RDC_PDAP_UART4, D1R | D1W),
 	RDC_PDAPn(RDC_PDAP_UART2, D0R | D0W),
 	RDC_PDAPn(RDC_PDAP_UART1, D0R | D0W),
+	RDC_PDAPn(RDC_PDAP_eCSPI2, D0R | D0W | D1R | D1W),
+	RDC_PDAPn(RDC_PDAP_GPIO3, D0R | D0W | D1R | D1W),
+	RDC_PDAPn(RDC_PDAP_GPIO5, D0R | D0W | D1R | D1W),
 
 	/* memory region */
 
